### PR TITLE
Add ycredit strategy

### PR DESCRIPTION
**Pull request by nourharidy**
Source complete.
Pending mainnet tests and review.

### CODE REVIEW

**Comment by nourharidy** [comment]
yCredit code: https://etherscan.io/address/0xe0839f9b9688a77924208ad509e29952dc660261#code

File: contracts/YCreditStrat.sol [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    constructor(IVault vault_, Timelock timelock_) {
+        vault = vault_;
+        timelock = timelock_;
+        yCredit = IYCredit(address(vault_.underlying())); // vault underlying must be yCredit
+    }
+
+    function invest() external override onlyVault {
+        reinvest();
```
Comment (0xKiwi) [inline contracts/YCreditStrat.sol:30]: We're staking and checking balances a redundant amount of times here. Is the cost worth the convenient internal function usage?

File: contracts/YCreditStrat.sol [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        uint balance = yCredit.balanceOf(address(this));
+        require(balance > 0);
+        yCredit.stake(balance);
+    }
+
+    function divest(uint amount) external override onlyVault {
+        reinvest();
+        yCredit.unstake(amount);
+        yCredit.transfer(address(vault), amount);
```
Comment (0xKiwi) [inline contracts/YCreditStrat.sol:39]: Also here, but more understandable. We can always check the balance after receiving rewards and then withdraw using that balance. But it's more risky, I don't feel strongly.

File: contracts/YCreditStrat.sol (hunk 1 above)
Comment (nourharidy) [reply contracts/YCreditStrat.sol:30]: You're right. I'll get rid of the function call and do it directly in each function

File: contracts/YCreditStrat.sol (hunk 2 above)
Comment (nourharidy) [reply contracts/YCreditStrat.sol:39]: I was thinking we need to determine the amount sent by the vault before it called `invest`. But in fact, it's not needed. Will remove the first `oldBalance` call.

**Review by 0xKiwi — APPROVED** [review]
LGTM!

**Pull request merged**